anymail/backends/unisender_go.py

```diff
@@ -0,0 +1,286 @@
+            **merge_global_data,
+        }
+
+    def set_anymail_id(self) -> None:
+        """Ensure each personalization has a known anymail_id for later event tracking"""
+        for recipient in self.data["recipients"]:
+            anymail_id = str(uuid.uuid4())
+
+            recipient.setdefault("metadata", {})
+            recipient["metadata"]["message_id"] = anymail_id
+
+            email_address = recipient["email"]
+            self.message_ids[email_address] = anymail_id
+
+    def set_from_email(self, email: EmailAddress) -> None:
+        self.data["from_email"] = email.addr_spec
+        self.data["from_name"] = email.display_name
+
+    def set_recipients(self, recipient_type: str, emails: list[EmailAddress]) -> None:
+        if not emails:
+            return
+        self.data["recipients"] = [
+            {"email": email.addr_spec, "substitutions": {"to_name": email.display_name}}
+            for email in emails
+        ]
+        self.all_recipients += emails
+
+    def set_subject(self, subject: str) -> None:
+        if subject != "":  # see note in set_text_body about template rendering
+            self.data["subject"] = subject
+
+    def set_reply_to(self, emails: list[EmailAddress]) -> None:
+        # Unisunder GO only supports a single address in the reply_to API param.
+        if len(emails) > 1:
+            self.unsupported_feature("multiple reply_to addresses")
+        if len(emails) > 0:
+            self.data["reply_to"] = emails[0].addr_spec
+
+    def set_extra_headers(self, headers: dict[str, str]) -> None:
+        self.data["headers"].update(headers)
+
+    def set_text_body(self, body: str) -> None:
+        if body == "":
+            return
+        if "body" not in self.data:
+            self.data["body"] = {}
+        self.data["body"]["plaintext"] = body
+
+    def set_html_body(self, body: str) -> None:
+        if body == "":
+            return
+        if "body" not in self.data:
+            self.data["body"] = {}
+        self.data["body"]["html"] = body
+
+    def add_attachment(self, attachment: Attachment) -> None:
+        att = {
+            "content": attachment.b64content,
+            "type": attachment.mimetype,
+            "name": attachment.name or "",  # required -- submit empty string if unknown
+        }
+        if attachment.inline:
+            self.data.setdefault("inline_attachments", []).append(att)
+        else:
+            self.data.setdefault("attachments", []).append(att)
+
+    def set_metadata(self, metadata: dict[str, str]) -> None:
+        self.data["global_metadata"] = metadata
+
+    def set_send_at(self, send_at: datetime.datetime) -> None:
+        self.data.setdefault("options", {})["send_at"] = send_at
+
+    def set_tags(self, tags: dict[str, str]) -> None:
+        self.data["tags"] = tags
+
+    def set_template_id(self, template_id: str) -> None:
+        self.data["template_id"] = template_id
```

Comment:
1. I moved extra fields to "set_esp_extra"
2. Added some fields there
3. I did NOT move template_id, because i saw set_template_id in some other ESPs, so i decided, it is OK. 

I dont check this conversation as resolved 'cause not sure, if you approve it.

Reply:
Oh, now I understand, thank you. Yeah, it's very nice and flexible, cool. I've been using the library configs kinda wrong way for a while xD